src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -643,4 +658,23 @@ export class NodeClientRequest extends ClientRequest {
     // @ts-ignore "agent" is a private property.
     this.agent?.destroy?.()
   }
+
+  private deferResponseEndUntil(
```

Comment:
Maybe premature but I'm creating a private method to help us reuse the same deferring logic for both mocked and original responses (the target `IncomingMessage` is different in case of both).

Reply:
I will likely repurpose this for the Socket interceptor. It would have its own implementation of awaiting the response listeners.